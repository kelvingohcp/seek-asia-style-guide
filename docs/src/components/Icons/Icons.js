import React from 'react';
import styles from './Icons.less';
import {
  PageBlock,
  Card,
  Section,
  Paragraph,
  Text
} from 'seek-asia-style-guide/react';
import Code from '../Demo/Code/Code';
import iconsSketchExports from '../../../../react/*/*.iconSketch.js';
import map from 'lodash/map';

const renderIcons = sketch => {
  return sketch && sketch.symbols ? (
    <div>
      <PageBlock>
        <div className={styles.symbols}>
          <Section className={styles.section}>
            {map(sketch.symbols || {}, (element, name) => (
              <div key={name} className={styles.symboleContainer}>
                <div className={styles.symbolName}>
                  <Text strong>{name.replace(/\//g, ' \u25B8 ')}</Text>
                </div>
                <div className={styles.symbolElement}>{element}</div>
                <div className={styles.codeStyle}>
                  <Code jsx={element} tenantPath="react" />
                </div>
              </div>
            ))}
          </Section>
        </div>
      </PageBlock>
    </div>
  ) : null;
};

const renderAllIcons = () => {
  return iconsSketchExports.map(symbols => {
    return renderIcons(symbols);
  });
};

export default function Icons() {
  return (
    <div>
      <PageBlock>
        <Section header>
          <Text screaming>Icons</Text>
        </Section>
        <Card transparent style={{ maxWidth: 720 }}>
          <Section>
            <Paragraph>
              <Text>
                These are the icons components available for use in the SEEK Asia style guide.
              </Text>
            </Paragraph>
          </Section>
        </Card>
      </PageBlock>

      <PageBlock>
        <Card className={styles.iconsCard}>{renderAllIcons()}</Card>
      </PageBlock>
    </div>
  );
}
