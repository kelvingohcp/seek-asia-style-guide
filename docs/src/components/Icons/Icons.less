@import (reference) "~seek-asia-style-guide/theme";

@iconWidth: @grid-base * 6;
@iconHeight: @grid-base * 6;

.symbolElement {
  svg {
    width: @iconWidth;
    height: @iconHeight;
  }
}

.section {
  display: flex;
  flex-wrap: wrap;
  border: 2px solid transparent;
  border-radius: 4px;
  transition: border-color .3s;

  @supports (display: grid) {
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(12rem, 1fr));
    padding: 0;
    grid-gap: 1rem;
    align-items: center;
    text-align: center;

    &:hover {
      border-color: #ddd;
    }
  }
}

.symbolContainer {
  padding: @grid-row-height * 3;
  display: flex;
  flex-flow: column-reverse;
  align-items: center;
}

.iconsCard {
  display: flex;
  flex-wrap: wrap;
  @supports (display: grid) {
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(12rem, 1fr));
    padding: 1rem;
    grid-gap: 2rem 1rem;
  }
}
