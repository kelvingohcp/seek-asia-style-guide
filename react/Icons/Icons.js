import React from 'react';
import PropTypes from 'prop-types';
import IconPack from './Icons.path';

const Icons = ({ iconName, ...restProps }) => {
  const getItem = IconPack.find(x => x.name === iconName);
  return (getItem) ? (
    <svg {...restProps} width="24" height="24" viewBox="0 0 100 100" preserveAspectRatio="xMidYMid meet">
      <title>{getItem.name}</title>
      <path d={getItem.path} fill="currentColor" />
    </svg>
  ) : null;
};

Icons.propTypes = {
  iconName: PropTypes.oneOf([
    'add',
    'advance',
    'alert',
    'application',
    'applicationHistory',
    'birthdate',
    'bonus',
    'bookmark',
    'calendar',
    'careerInsight',
    'careerLevel',
    'certifications',
    'check',
    'comments',
    'company',
    'companyProfile',
    'currentStatus',
    'dentalInsurance',
    'doublePay',
    'down',
    'download',
    'edit',
    'education',
    'employer',
    'employmentType',
    'exit',
    'expand',
    'experience',
    'expired',
    'extend',
    'facebook',
    'featured',
    'fiveDayWork',
    'flexibleWorkingHours',
    'freeShuttleBus',
    'gratuity',
    'hidden',
    'home',
    'housingAllowance',
    'information',
    'instagram',
    'jobAd',
    'jobAlert',
    'jobFunction',
    'jobInvitation',
    'language',
    'lifeInsurance',
    'location',
    'locked',
    'mail',
    'medical',
    'menu',
    'myJobsDB',
    'negative',
    'overtimePay',
    'pastApplication',
    'positive',
    'preferences',
    'previous',
    'profileFemale',
    'profileMale',
    'promotion',
    'purchaseCredits',
    'RC',
    'refresh',
    'remove',
    'resources',
    'resume',
    'rss',
    'salary',
    'search',
    'settings',
    'share',
    'show',
    'sivaRC',
    'skills',
    'talentSearch',
    'telephone',
    'timestamp',
    'top',
    'transportAllowance',
    'travelAllowance',
    'twitter',
    'unlocked',
    'upload',
    'youtube'
  ]).isRequired
};

Icons.defaultProps = {
  iconName: 'birthdate'
};

export default Icons;
