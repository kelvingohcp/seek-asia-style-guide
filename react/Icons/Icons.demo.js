import React from 'react';
import Icons from './Icons';
import IconPack from './Icons.path';
import PropTypes from 'prop-types';
import { PageBlock } from 'seek-asia-style-guide/react';

const IconContainer = ({ component: DemoComponent, componentProps }) => {
  return (
    <PageBlock style={{ width: '100%' }}>
      <DemoComponent {...componentProps} />
    </PageBlock>
  );
};

IconContainer.propTypes = {
  component: PropTypes.func,
  componentProps: PropTypes.object
};

const getSvgPaths = IconPack.map(obj => {
  const rObj = {
    label: obj.name,
    transformProps: props => ({
      ...props,
      iconName: obj.name
    })
  };
  return rObj;
});
export default {
  route: '/icons',
  title: 'Icons',
  component: Icons,
  initialProps: {
    iconName: 'birthdate'
  },
  options: [
    {
      label: 'Type',
      type: 'radio',
      states: getSvgPaths
    }
  ]
};
