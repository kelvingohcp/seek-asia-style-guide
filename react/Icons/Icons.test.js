import React from 'react';
import { shallow } from 'enzyme';

import Icons from './Icons';

describe('Icons', () => {
  const defaultProps = {
    iconName: 'birthdate'
  };

  it('renders with default props', () => {
    const wrapper = shallow(<Icons {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with title', () => {
    const wrapper = shallow(<Icons title='title' {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
