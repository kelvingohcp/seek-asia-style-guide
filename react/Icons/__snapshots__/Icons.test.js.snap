// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons renders with default props 1`] = `
<Icon
  className=""
  markup="<svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 100 100\\" preserveAspectRatio=\\"xMidYMid meet\\">
      <title>birthdate</title>
      <path d=\\"M2.2 47.4v-7.8h95.7v7.8H2.2zm78.2 11.7c2.6-1.3 5.2-2.2 8.7-2.2h8.7v31.3c0 6.5-5.7 11.7-13 11.7H15.2c-7 0-13-5.2-13-11.7V56.5h8.7c3.5 0 6.1 1.3 8.7 2.2 1.7.9 3.5 1.7 4.3 1.7s2.6-.9 4.3-1.7c2.6-.9 5.7-2.2 8.7-2.2 3 0 6.1 1.3 8.7 2.2 1.7.9 3.5 1.7 4.3 1.7s3-.9 4.3-1.7c2.6-1.3 5.2-2.2 8.7-2.2 3.5 0 6.1 1.3 8.7 2.2 1.7.9 3.5 1.7 4.3 1.7s3.2-.4 4.5-1.3zM15.2 33.5v-20h10v20h-10zM20 10c-2.6 0-4.8-2.2-4.8-5.2C15.2 2.2 17.4 0 20 0c2.6 0 4.8 2.2 4.8 4.8.4 3-1.8 5.2-4.8 5.2zm25.2 23.5v-20h10v20h-10zM50 10c-2.6 0-4.8-2.2-4.8-4.8 0-3 2.2-5.2 4.8-5.2 2.6 0 4.8 2.2 4.8 4.8 0 3-2.2 5.2-4.8 5.2zm24.8 23.5v-20h10v20h-10zM80 10c-2.6 0-4.8-2.2-4.8-4.8 0-2.6 2.2-4.8 4.8-4.8 2.6 0 4.8 1.7 4.8 4.3 0 3.1-2.2 5.3-4.8 5.3z\\" fill=\\"currentColor\\" />
    </svg>"
  svgClassName=""
/>
`;

exports[`Icons renders with title 1`] = `
<Icon
  className=""
  markup="<svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 100 100\\" preserveAspectRatio=\\"xMidYMid meet\\">
      <title>title</title>
      <path d=\\"M2.2 47.4v-7.8h95.7v7.8H2.2zm78.2 11.7c2.6-1.3 5.2-2.2 8.7-2.2h8.7v31.3c0 6.5-5.7 11.7-13 11.7H15.2c-7 0-13-5.2-13-11.7V56.5h8.7c3.5 0 6.1 1.3 8.7 2.2 1.7.9 3.5 1.7 4.3 1.7s2.6-.9 4.3-1.7c2.6-.9 5.7-2.2 8.7-2.2 3 0 6.1 1.3 8.7 2.2 1.7.9 3.5 1.7 4.3 1.7s3-.9 4.3-1.7c2.6-1.3 5.2-2.2 8.7-2.2 3.5 0 6.1 1.3 8.7 2.2 1.7.9 3.5 1.7 4.3 1.7s3.2-.4 4.5-1.3zM15.2 33.5v-20h10v20h-10zM20 10c-2.6 0-4.8-2.2-4.8-5.2C15.2 2.2 17.4 0 20 0c2.6 0 4.8 2.2 4.8 4.8.4 3-1.8 5.2-4.8 5.2zm25.2 23.5v-20h10v20h-10zM50 10c-2.6 0-4.8-2.2-4.8-4.8 0-3 2.2-5.2 4.8-5.2 2.6 0 4.8 2.2 4.8 4.8 0 3-2.2 5.2-4.8 5.2zm24.8 23.5v-20h10v20h-10zM80 10c-2.6 0-4.8-2.2-4.8-4.8 0-2.6 2.2-4.8 4.8-4.8 2.6 0 4.8 1.7 4.8 4.3 0 3.1-2.2 5.3-4.8 5.3z\\" fill=\\"currentColor\\" />
    </svg>"
  svgClassName=""
/>
`;
